src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -5,28 +5,83 @@
 import * as React from 'react';
 import flow from 'lodash/flow';
 import { FormattedMessage } from 'react-intl';
+
+import { Text, InlineError, LoadingIndicator } from '@box/blueprint-web';
+import { AddMetadataTemplateDropdown } from '@box/metadata-editor';
+
+import API from '../../api';
+import { type ElementsXhrError } from '../../common/types/api';
+import { type ElementOrigin } from '../common/flowTypes';
 import { withAPIContext } from '../common/api-context';
 import { withErrorBoundary } from '../common/error-boundary';
 import { withLogger } from '../common/logger';
 import { ORIGIN_METADATA_SIDEBAR_REDESIGN } from '../../constants';
 import { EVENT_JS_READY } from '../common/logger/constants';
 import { mark } from '../../utils/performance';
+
 import messages from '../common/messages';
 
+import { MetadataTemplate } from '../../common/types/metadata';
+import { type WithLoggerProps } from '../../common/types/logging';
+
 import './MetadataSidebarRedesign.scss';
+import useSidebarMetadataFetcher, { STATUS } from './hooks/useSidebarMetadataFetcher';
 
 const MARK_NAME_JS_READY = `${ORIGIN_METADATA_SIDEBAR_REDESIGN}_${EVENT_JS_READY}`;
 
 mark(MARK_NAME_JS_READY);
 
-function MetadataSidebarRedesign() {
+export interface ExternalProps {
+    isFeatureEnabled: boolean;
+}
+
+interface PropsWithoutContext extends ExternalProps {
+    elementId: string;
+    fileId: string;
+    hasSidebarInitialized?: boolean;
+}
+
+export interface ErrorContextProps {
+    onError: (error: ElementsXhrError | Error, code: string, contextInfo?: Object, origin?: ElementOrigin) => void;
+}
+
+export interface MetadataSidebarRedesignProps extends PropsWithoutContext, ErrorContextProps, WithLoggerProps {
+    api: API;
+}
+
+function MetadataSidebarRedesign({ api, fileId, onError, isFeatureEnabled }: MetadataSidebarRedesignProps) {
+    const [selectedTemplates, setSelectedTemplates] = React.useState<Array<MetadataTemplate>>([]);
+
+    const { templates, errorMessage, status } = useSidebarMetadataFetcher(api, fileId, onError, isFeatureEnabled);
+
+    const renderMetadataDropdown = status === STATUS.SUCCESS && templates && (
+        <AddMetadataTemplateDropdown
+            availableTemplates={templates}
+            selectedTemplates={selectedTemplates}
+            onSelect={(selectedTemplate): void => {
+                setSelectedTemplates([...selectedTemplates, selectedTemplate]);
+            }}
+        />
+    );
+
+    const renderErrorMessage = status === STATUS.ERROR && errorMessage && (
+        <InlineError>
+            <FormattedMessage {...errorMessage} />
+        </InlineError>
+    );
+
     return (
-        <div className="bcs-MetadataSidebarRedesign">
-            <h3>
-                <FormattedMessage {...messages.sidebarMetadataTitle} />
-            </h3>
-            <hr />
-            <p>Hello from Metadata Sidebar redesign</p>
+        <div className="bcs-MetadataSidebarRedesign bcs-content">
```

Comment:
what's the longterm plan? will this be refactored to use `SidebarContent` or is that component going away?

Reply:
Good catch! I've changed implementation to use `SidebarContent`